### PR TITLE
consolidate configuration

### PR DESCRIPTION
**Pull request by prybalko**
- remove single feed mode
- consolidate configuration while keeping backward compatibility

see #70

### CODE REVIEW

**Comment by umputun** [comment]
I have found use-cases for single-feed configuration and we can't drop it yet

**Comment by prybalko** [comment]
Just curious, that's the use case? It seems to be just a special case of multi-feed configuration.

Having a no-conf mode raises the question of where to store the telegram config. It doesn't use the `yaml` config, so it has to be done via env variables or cli args. But in this case, we end up where we started and this ticket doesn't make sense anymore. Or it's ok to have 2 ways for configuration: yml file and cil/env options?

**Comment by umputun** [comment]
the use case is kind of strange - to publish updates on feed change to both tg and twitter. This is how some of my podcasts do it. The reason why I have added this mode back in days - I wanted as simple as possible thingy to be fully defined in the docker-compose.yml  (via the environment) without a need to maintain one more configuration file. 

Probably we can go with a breaking change here and enforce the configuration yml file for those cases too.

File: app/config/config.go [hunk 1]
```diff
@@ -16,6 +17,9 @@ import (
 type Conf struct {
 	Feeds  map[string]Feed `yaml:"feeds"`
 	System struct {
+		DB             string        `yaml:"db"`
+		AdminPasswd    string        `yaml:"admin-passwd"`
+		Dbg            bool          `yaml:"debug"`
```
Comment (umputun) [inline app/config/config.go:22]: I think dbg should stay as a parameter

File: app/config/config.go [hunk 2]
```diff
@@ -35,6 +39,21 @@ type Conf struct {
 		FilesLocation   string             `yaml:"files_location"`
 		RSSLocation     string             `yaml:"rss_location"`
 	} `yaml:"youtube"`
+
+	Telegram struct {
+		Server  string        `yaml:"server"`
+		Channel string        `yaml:"channel"`
```
Comment (umputun) [inline app/config/config.go:45]: this one is unclear. Used to be a part of single-feed setup, but if we drop it chanel is irrelevant here, as they defined per feed

File: app/config/config.go [hunk 3]
```diff
@@ -35,6 +39,21 @@ type Conf struct {
 		FilesLocation   string             `yaml:"files_location"`
 		RSSLocation     string             `yaml:"rss_location"`
 	} `yaml:"youtube"`
+
+	Telegram struct {
```
Comment (umputun) [inline app/config/config.go:43]: probably both telegram and twitter should be under System, maybe even System.Notifications

**Review by umputun — CHANGES_REQUESTED** [review]

File: app/config/config.go (hunk 2 above)
Comment (prybalko) [reply app/config/config.go]: My mistake, thanks for catching this.

File: app/config/config.go (hunk 3 above)
Comment (prybalko) [reply app/config/config.go]: Done

**Comment by prybalko** [comment]
Is this PR still needed? Any comments/suggestions?

**Comment by umputun** [comment]
still not decided yet, need time to see what it looks like to switch a real single-mode setup to the  full one

**Pull request closed without merging**